azurekeyvault-flexvolume/oauth.go

```diff
@@ -191,6 +196,39 @@ func GetServicePrincipalToken(tenantId string, env *azure.Environment, resource
 	return nil, fmt.Errorf("no credentials provided for AAD application %s", aADClientID)
 }
 
+func retryFetchToken(req *http.Request, maxAttempts int) (resp *http.Response, err error) {
+	attempt := 0
+	delay := time.Duration(0)
+	client := &http.Client{}
+	const maxDelay time.Duration = 60 * time.Second
+
+	// retry guidance as suggested here - https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#retry-guidance
+	for attempt < maxAttempts {
+		resp, err = client.Do(req)
```

Comment:
Are there really non-retriable errors like nmi is not reachable at all or some other errors ? During the initial deployment this can be an issue, but since then this might be a fatal error which we will retry for the whole time gap before declaring the failure.